discovery-provider/src/utils/session_manager.py

```diff
@@ -11,10 +14,24 @@
 class SessionManager:
     def __init__(self, db_url, db_engine_args):
         self._engine = create_engine(db_url, **db_engine_args)
+
+        @event.listens_for(Engine, "before_cursor_execute", retval=True)
```

Comment:
if I understand correctly listens_for is just a decorator around listen like the usage below. Should we standardize?

Reply:
Yea I'll change this like the others.